src/instantsearch/instantsearch.ts

```diff
@@ -135,8 +135,57 @@ export type SearchResponse = {
   index?: string;
 };
 
+type HitAttributeHighlightResult = {
+  value: string;
+  matchLevel: 'none' | 'partial' | 'full';
+  matchedWords: string[];
+  fullyHighlighted?: boolean;
+};
+
+type HitHighlightResult = {
+  [attribute: string]:
+    | HitAttributeHighlightResult
+    | HitAttributeHighlightResult[]
+    | HitHighlightResult;
+};
+
+type HitAttributeSnippetResult = Pick<
+  HitAttributeHighlightResult,
+  'value' | 'matchLevel'
+>;
+
+type HitSnippetResult = {
+  [attribute: string]:
+    | HitAttributeSnippetResult
+    | HitAttributeSnippetResult[]
+    | HitSnippetResult;
+};
+
 export type Hit = {
-  _highlightResult?: object;
+  [attribute: string]: any;
+  objectID: string;
+  _highlightResult?: HitHighlightResult;
+  _snippetResult?: HitSnippetResult;
+  _rankingInfo?: {
+    promoted: boolean;
+    nbTypos: number;
+    firstMatchedWord: number;
+    proximityDistance?: number;
+    geoDistance: number;
+    geoPrecision?: number;
+    nbExactWords: number;
+    words: number;
+    filters: number;
+    userScore: number;
+    matchedGeoLocation?: {
+      lat: number;
+      lng: number;
+      distance: number;
+    };
+  };
+  _distinctSeqID?: number;
+  __position: number;
+  __queryID?: string;
```

Comment:
are these last two also used like this in Angular InstantSearch?

Reply:
They're not but an Angular InstantSearch user might need them, typically in a template when trying to pass it in a URL.

```html
<ais-hits>
  <ng-template let-hits="hits">
    <a 
       *ngFor="let hit of hits"
       href="product.html?queryID={{hit.__queryID}}"
     >
        ...
    </a>
  </ng-template>
</ais-hits>
```

Other than that, I don't want to start diverging for the types we have in InstantSearch. Does that make sense?

Reply:
makes sense, my question was more _if_  this property is available, which seems to be true

Reply:
ah yes it is, it's exposed by connectHits and connectInfiniteHits